lib/linters/enhanced_idv_events_linter.rb

```diff
@@ -0,0 +1,90 @@
+        extend AutoCorrector
+
+        RESTRICT_ON_SEND = [:track_event]
+
+        ENHANCED_ARGS = [:proofing_components].freeze
+
+        def on_send(track_event_send)
+          method = track_event_send.each_ancestor(:def).first
+          return if !should_check_method?(method.method_name)
+
+          ENHANCED_ARGS.each do |arg_name|
+            check_arg_present_on_event_method(arg_name, method)
+            check_arg_present_in_track_event_send(arg_name, track_event_send)
+          end
+        end
+
+        private
+
+        def check_arg_present_in_track_event_send(arg_name, track_event_send)
+          # We expect there is a hash argument that includes arg_name
+          present = track_event_send.arguments.any? do |arg|
+            next unless arg.type == :hash
+
+            arg.pairs.any? do |pair|
+              key_ok = pair.key.type == :sym && pair.key.value == arg_name
+              value_ok = pair.value.type == :lvar && pair.node_parts[0].value == arg_name
+              key_ok && value_ok
+            end
+          end
+
+          if !present
+            message = "#{arg_name} is missing from track_event call."
+            add_offense(track_event_send, message:) do |corrector|
+              correct_arg_missing_from_track_event_send(arg_name, track_event_send, corrector)
+            end
+          end
+        end
+
+        def check_arg_present_on_event_method(arg_name, method)
+          arg = method.arguments.find { |a| a.name == arg_name }
+          return if arg
+
+          add_offense(method, message: "Method is missing #{arg_name} argument.") do |corrector|
+            correct_arg_missing_from_event_method(arg_name, method, corrector)
+          end
+        end
+
+        def correct_arg_missing_from_event_method(arg_name, method, corrector)
+          arg = method.arguments.find { |a| a.kwarg_type? && a.name == arg_name }
+          return if arg
+
+          kwrest = method.arguments.find { |a| a.kwrestarg_type? }
+          return if !kwrest
+
+          corrector.insert_before(kwrest, "#{arg_name}: nil, ")
+        end
+
+        def correct_arg_missing_from_track_event_send(arg_name, track_event_send, corrector)
+          hash_arg = track_event_send.arguments.find { |arg| arg.hash_type? }
+          return unless hash_arg
+
+          return if hash_arg.pairs.any? { |pair| pair.key.value == arg_name }
+
+          # Put the reference into the hash, before the splat
+          kwsplat = hash_arg.children.find { |child| child.kwsplat_type? }
+          if kwsplat
+            corrector.insert_before(kwsplat, "#{arg_name}: #{arg_name}, ")
+          end
+        end
+
+        def events_enhancer_class
+          # Idv::AnalyticsEventsEnhancer keeps its own record of which
+          # methods it wants to "enhance".
+          require_relative '../../app/services/idv/analytics_events_enhancer'
+          ::Idv::AnalyticsEventsEnhancer
+        rescue
+          nil
```

Comment:
seems like we should let this error if it can't load the class? its already tied so closely to our codebase?